src/sdam/topology.ts

```diff
@@ -563,24 +569,10 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
         new ServerSelectionStartedEvent(selector, this.description, options.operationName)
       );
     }
-    const serverSelectionTimeoutMS = options.serverSelectionTimeoutMS ?? 0;
-    let timeout: Timeout | null;
-    if (options.timeout) {
-      // CSOT Enabled
-      if (options.timeout.duration > 0 || serverSelectionTimeoutMS > 0) {
-        if (
-          options.timeout.duration === serverSelectionTimeoutMS ||
-          csotMin(options.timeout.duration, serverSelectionTimeoutMS) < serverSelectionTimeoutMS
-        ) {
-          timeout = options.timeout;
-        } else {
-          timeout = Timeout.expires(serverSelectionTimeoutMS);
-        }
-      } else {
-        timeout = null;
-      }
-    } else {
-      timeout = Timeout.expires(serverSelectionTimeoutMS);
+    let timeout;
+    if (options.timeoutContext) timeout = options.timeoutContext.serverSelectionTimeout;
+    else {
+      timeout = Timeout.expires(options.serverSelectionTimeoutMS ?? 0);
```

Comment:
This is (or should be) dead code

Reply:
which part in particular is dead code?  The else block?